packages/xsnap/src/xsnap.js

```diff
@@ -89,6 +89,8 @@ export function xsnap(options) {
   /** @type {Deferred<void>} */
   const vatExit = defer();
 
+  assert(!/^-/.test(name), `name '${name}' cannot start with hyphen`);
```

Comment:
Since this is an assertion that could fail at runtime, do you want to tailor the regexp used by `vat-admin/vat-vat-admin.js` to also insist on an initial non-hyphen character?

Reply:
I thought about that, but the unavoidable-by-userspace `${vatID}` prefix means that `createVat(bcap, { name: '--sneaky' })` will result in xsnap.js getting `name: 'v123:--sneaky'`, which seems fine.

Reply:
In that case, could you call this variable something different than `name`, if it really is `${vatID}:${name}`?

Reply:
Hm, that would be a breaking change to the `xsnap` API (because `name` appears in an options bag, not as a positional argument), and in theory we might have non-swingset users of `@agoric/xsnap`. These two uses of `name` appear in different packages.

Would it look better if `manager-subprocess-xsnap.js` did:
```js
const {
  ..
  name: vatName,
} = managerOptions;
..
const argName = `${vatID}:${vatName !== undefined ? vatName}`;
```
instead?

Reply:
Yes, that would be great.  I was just misled by the reuse of the identifier `name` when it had different contents.